test/unit/ui_notifications/cache_handler_test.rb

```diff
@@ -0,0 +1,34 @@
+require 'test_helper'
+
+class UINotificationsCacheHandlerTest < ActiveSupport::TestCase
+  test 'notification should raise without a subject' do
+    assert_raise(Foreman::Exception) { UINotifications::Base.new(nil) }
+  end
+
+  test 'should provide JSON payload' do
+    response = JSON.parse(UINotifications::CacheHandler.new(User.first.id).payload)
+    assert_equal({"notifications"=>[]}, response)
+  end
+
+  test 'should have a unique cache key per user' do
+    assert_equal 'notification-1', UINotifications::CacheHandler.new(1).send(:cache_key)
+  end
+
+  test 'should use default cache expiry time' do
+    expiry = UINotifications::CacheHandler.new(1).send(:cache_expiry)
+    assert_equal 1.hour, expiry
+  end
+
+  test 'should calculate cache expiry time based on the notification expiry' do
+    user = FactoryGirl.create(:user)
+    blueprint = FactoryGirl.create(:notification_blueprint, :expires_in => 3.hours,
+                              :message => 'this test just executed successfully')
+    notification = FactoryGirl.create(:notification, :notification_blueprint => blueprint, :audience => 'global')
+    FactoryGirl.create(:notification_recipient, :notification =>  notification, user_id: user.id)
```

Comment:
NP: I'm not opposed to new-style hashes, but at least be consistent with one style per line :smile: